selfservice/strategy/oidc/provider_vk.go

```diff
@@ -0,0 +1,128 @@
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+
+	"github.com/ory/herodot"
+)
+
+type ProviderVK struct {
+	config *Configuration
+	public *url.URL
+}
+
+func NewProviderVK(
+	config *Configuration,
+	public *url.URL,
+) *ProviderVK {
+	return &ProviderVK{
+		config: config,
+		public: public,
+	}
+}
+
+func (g *ProviderVK) Config() *Configuration {
+	return g.config
+}
+
+func (g *ProviderVK) oauth2() *oauth2.Config {
+	return &oauth2.Config{
+		ClientID:     g.config.ClientID,
+		ClientSecret: g.config.ClientSecret,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  "https://oauth.vk.com/authorize",
+			TokenURL: "https://oauth.vk.com/access_token",
+		},
+		Scopes:      g.config.Scope,
+		RedirectURL: g.config.Redir(g.public),
+	}
+}
+
+func (g *ProviderVK) AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption {
+	return []oauth2.AuthCodeOption{}
+}
+
+func (g *ProviderVK) OAuth2(ctx context.Context) (*oauth2.Config, error) {
+	return g.oauth2(), nil
+}
+
+func (g *ProviderVK) Claims(ctx context.Context, exchange *oauth2.Token) (*Claims, error) {
+
+	o, err := g.OAuth2(ctx)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	client := o.Client(ctx, exchange)
+
+	u, err := url.Parse("https://api.vk.com/method/users.get?fields=photo_200,nickname,bdate,sex&access_token=" + exchange.AccessToken + "&v=5.103")
```

Comment:
Looks like these fields are something people would like to configure themselves?

Reply:
Most of the fields do not match claims.
https://vk.com/dev/fields 
https://www.ory.sh/kratos/docs/concepts/credentials/openid-connect-oidc-oauth2/#external-variable-claims